src/main/java/io/pravega/connectors/flink/source/reader/PravegaRecordEmitter.java

```diff
@@ -58,15 +62,20 @@ public PravegaRecordEmitter(DeserializationSchema<T> deserializationSchema) {
 
     @Override
     public void emitRecord(EventRead<ByteBuffer> record, SourceOutput<T> output, PravegaSplit state) throws Exception {
-        // If the record to emit is a checkpoint, record the checkpoint ID for later
-        if (record.isCheckpoint()) {
+        if (record.isCheckpoint()) { // If the record to emit is a checkpoint, record the checkpoint ID for later
```

Comment:
I would recommend this to be in the method javadoc, and "for later" can be improved as "for reporting back to the source reader".